### ISSUE TITLE
Extend SC2086 to warn against unquoted executable paths

### ISSUE THREAD

**Issue by timothyandrew** [koalaman/shellcheck#1622]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2086
- My shellcheck version (`shellcheck --version` or "online"): 0.6.0
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```bash

#!/usr/bin/env bash
PATH='/Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin':$PATH
EXECUTABLE=$(command -v jarsigner)

# $EXECUTABLE is now set to '/Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/jarsigner'

$EXECUTABLE -help 
```

#### Here's what shellcheck currently says:
No output, return code 0.

#### Here's what I wanted or expected to see:
A warning against unquoted usage of $EXECUTABLE; this small script fails on my machine with `bash: line 5: /Applications/Android: No such file or directory`, but works once I change the last line to `"$EXECUTABLE" -help`.
